### DISCUSSION TITLE
clangd doesn't find std header?

### DISCUSSION THREAD

**Discussion by vuongdanghuy** [clangd/clangd#1497] [Q&A] [answered]
Hi everyone.

I'm using coc-clangd with vim but somehow clangd does not find any of the std header. Here is the clangd log in my local machine.
![image](https://user-images.githubusercontent.com/25819962/218241233-c1d8a688-42d0-4920-8542-6e12b008e4c2.png)

I do not understand why, because it work great in my Docker container as shown here.
![image](https://user-images.githubusercontent.com/25819962/218241277-263944f8-71c5-4217-96fa-0541aac740c1.png)

Any idea why it happen and how to fix it?

Thanks a lot.

**Comment by HighCommander4** [comment, marked as answer]
Please have a look at https://clangd.llvm.org/troubleshooting#cant-find-standard-library-headers-map-stdioh-etc

If you're not able to get it to work using the suggestions there, please add `--log=verbose` to clangd's command line arguments, and share the logs in text form (not screenshots)

**Reply by vuongdanghuy** [reply]
Thanks.

After setting --log=verbose, I see that cc1 is searching headers in /usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++ (or /usr/include/c++ for short) meanwhile standard lib headers is in /usr/include/c++/11. 

It seem that i have gcc-12 installed in /usr/lib/gcc/x86_64-linux-gnu but it does not come with headers in /usr/include/c++/12. After remove gcc-12 now it worked normally.

**Reply by HighCommander4** [reply]
This is a known issue affecting Ubuntu 22.04, there is some further discussion of it at https://github.com/clangd/clangd/issues/1394, but yeah uninstalling `gcc-12` (or installing `g++-12` which installs the version 12 stdlib headers) is the easiest workaround.
